client/src/components/v2/OverlayMessage.js

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback } from 'react';
+import { Button, Card, FormControl, InputGroup } from 'react-bootstrap';
+import { useMessageState } from 'api/v2/camera';
+
+/**
+ * Text field for sending messages to the rider via the active overlay.
+ *
+ * @returns {React.Component} Component
+ */
+export default function OverlayMessage() {
+  const { message, setMessage, sendMessage } = useMessageState();
```

Comment:
To keep the API "pure" it's preferable to keep state outside of the API where possible. An alernative interface could be `sendMessage(message)" and keep the current message state in the OverlayMessage component instead

Reply:
Something like this?

(wow, this PR was so overcomplicated)